src/NzbDrone.Core/Music/RefreshTrackService.cs

```diff
@@ -37,12 +37,17 @@ public void RefreshTrackInfo(Album album)
 
             foreach (var release in album.AlbumReleases.Value)
             {
-
-                var existingTracks = _trackService.GetTracksByForeignReleaseId(release.ForeignReleaseId);
+                var dupeFreeRemoteTracks = release.Tracks.Value.DistinctBy(m => new { m.ForeignTrackId, m.TrackNumber }).ToList();
+                
+                // Search both ways to make sure we properly deal with tracks that have been moved from one release to another
+                // as well as deleting any tracks that have been removed from a release.
+                // note that under normal circumstances, a track would be captured by both queries.
+                var existingTracksByRelease = _trackService.GetTracksByForeignReleaseId(release.ForeignReleaseId);
+                var existingTracksById = _trackService.GetTracksByForeignTrackIds(dupeFreeRemoteTracks.Select(x => x.ForeignTrackId).ToList());
```

Comment:
I don't think tracks can move from one release to another like releases can in MB can they? You would have to delete the track on one release during edit and add a new track to the other release.

Reply:
I'm not sure, I've never edited anything on MB. But this [sentry issue](https://sentry.io/lidarr/lidarr/issues/351278015/?referrer=github_plugin) has lots of events where track foreign Id unique constraint is violated and I'm not sure how else that would happen.